app/api/v_sphere/folder.rb

```diff
@@ -83,7 +90,11 @@ def create_vm(cpu, ram, capacity, name, cluster)
 
       vm_config = creation_config(cpu, ram, capacity, add_prefix(name), cluster.networks.first)
       vm = @folder.CreateVM_Task(config: vm_config, pool: cluster.resource_pool).wait_for_completion
-      VSphere::VirtualMachine.new vm
+      VSphere::VirtualMachine.new vm, folder: self
+    end
+
+    def equal?(other)
```

Comment:
`equal?` has a specific purpose in Ruby, i.e. to

> determine object identity (that is, a.equal?(b) if and only if a is the same object as b)

(http://ruby-doc.org/core-2.6.2/Object.html#method-i-eql-3F)

Therefore "Unlike ==, the equal? method should never be overridden by subclasses"

Most likely, `eql?` would be a better fit here

Reply:
changed it to eql?